File: java/grape-sdk/src/test/java/com/alibaba/grape/ds/BitsetTest.java
package com.alibaba.grape.ds;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class BitsetTest {

    private Bitset bitset;

    @Before
    public void before() {
        bitset = Bitset.factory.create();
        bitset.init(20);
    }

    @Test
    public void test1() {
        for (int i = 0; i < 10; i += 2) {
            bitset.set_bit((long) i);
        }
        for (int i = 0; i < 10; i += 2) {
            Assert.assertTrue(bitset.get_bit((long) i));
        }
        for (int i = 1; i < 10; i += 2) {
            Assert.assertFalse(bitset.get_bit((long) i));
        }
    }
}

File: analytical_engine/java/graphx-on-graphscope/src/test/scala/com/alibaba/graphscope/graphx/BitSetTest.scala
package com.alibaba.graphscope.graphx

import com.alibaba.graphscope.graphx.utils.BitSetWithOffset
import org.apache.spark.util.collection.BitSet
import org.junit.runner.RunWith
import org.scalatest.FunSuite
import org.scalatest.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class BitSetTest extends FunSuite{
  test("test1"){
    val bitSet = new BitSet(8546613)
    bitSet.setUntil(8546612)
    var ind = 0;
    val time0 = System.nanoTime()
    while (ind > 0){
      ind = bitSet.nextSetBit(ind + 1)
    }
    val time1 = System.nanoTime()
    System.out.println(s"Bitset iter cost ${(time1 - time0) / 1000000} ms")
  }

  test("test2"){
    val bitSet = new BitSet(1000)
    bitSet.setUntil(1000)
    assert(bitSet.cardinality() == 1000)
    assert(bitSet.get(999))
  }

  test("test3"){
    val myBitSet = new BitSetWithOffset(500,502)
    assert(myBitSet.cardinality() == 0)
    assert(myBitSet.bitset.cardinality() == 0)
    myBitSet.set(500)
    assert(myBitSet.get(500))
    assert(!myBitSet.get(501))
    myBitSet.set(501)
    assert(myBitSet.get(501))
  }
}
